api/config.go

```diff
@@ -0,0 +1,121 @@
+package api
+
+import (
+	"encoding/base64"
+	"fmt"
+	"gopkg.in/yaml.v2"
+	"io/ioutil"
+	"os"
+	"regexp"
+	"strings"
+)
+
+var Cfg *Config
+
+type Config struct {
+	Backend  string            `yaml:"backend"`
+	Backends map[string]string `yaml:"backends"`
+	Aliases  map[string]string `yaml:"aliases"`
+	Path     string            `yaml:"-"` // omit this little guy
+}
+
+func LoadConfig(p string) error {
+	Cfg = &Config{
+		Path:     p,
+		Backends: map[string]string{},
+		Aliases:  map[string]string{},
+	}
+	if _, err := os.Stat(p); os.IsNotExist(err) {
+		return nil
+	}
+
+	data, err := ioutil.ReadFile(p)
+	if err != nil {
+		return err
+	}
+
+	err = yaml.Unmarshal(data, Cfg)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (cfg *Config) Save() error {
+	data, err := yaml.Marshal(cfg)
+	if err != nil {
+		return err
+	}
+
+	err = ioutil.WriteFile(cfg.Path, data, 0644)
+	if err != nil {
+		return err
+	}
```

Comment:
Change this to 0600, so that auth credentials are not visible to other users, or other users in the same group (for Jumpboxen without sudo).

Also, is there any way to do this as a 2-stage creat+rename?  That way we always creat the file with 0600, and we never put creds on-disk with open permissions (in case the user was silly and chmod'd the file)